Workaround missing parameters from job schedule

QBY-MarkusMaring opened the pull request:
# Description

<!-- Please include a summary of changes and which issues are fixed -->
<!-- Example -->

When updating the automation runbook in any way azurerm will allways destroy/create the job schedule: https://github.com/hashicorp/terraform-provider-azurerm/blob/12ea051565bbcd42ec68d6260a42d8961feb84e8/internal/services/automation/automation_runbook_resource.go#L329C1-L356C3

This will ignore any previous parameters on that schedule. To make sure they are not ignored this PR forces the replacement of the job schedule resource when updating the runbook.

# Change overview (tick true):

- [ ] This introduces backward incompatible changes
- [ ] This adds a new backward compatible Feature
- [x] This fixes a Bug

# Version information: 

<!-- Look up the current/previous Version and update it below -->
- Previous Version: `2.0.2`
<!-- Update the version below, under which version you plan to release this PR. See Link on information how to increment the version number -->
- Next Version based on [Semantic Versioning](https://semver.org/#summary) (see above): `2.0.3`

# How Has This Been Tested?

<!-- Please list and describe tests that you performed -->
<!-- You should always do some tests! -->

- [x] Tested in test environment PCMS DEV1

# Checklist:

- [x] I have run tests and documented them above
- [x] I have performed a self-review of my own code
- [ ] I have updated the documentation
- [x] I have updated the CHANGELOG

QBY-ThimoLimpert commented:
Wow, interesting finding. I didn't look at the implementation in the provider. But that sounds like a bug to me. Do you want to open an issue there?

QBY-MarkusMaring commented:
> Wow, interesting finding. I didn't look at the implementation in the provider. But that sounds like a bug to me. Do you want to open an issue there?

There was an issue for it. Then someone posted the workaround, so the azurerm team decided that this is now the solution and closed the issue as "not planned"...



> * Shouldn't be the next version `2.0.3`if it only fixes a bug?
> * Why didn't you change the Changelog?
> * Should we mention somewhere, that changing the runbook recreates the schedule?

- fixed
- fixed
- Added the issue link in a comment
